spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1314,6 +1314,10 @@ def tab_name_editor(self):
     @Slot(object, object)
     def edit_file(self, filename, line):
         """Handle %edit magic petitions."""
+        if not osp.isfile(filename):
```

Comment:
Perhaps we should surround this with a try/except in case people try to run this in a remote kernel? I mean, since both local and remote filesystems are different, this makes little sense.